3-UsingAPIs/Week1/homework/ex1-johnWho.js

```diff
@@ -9,21 +10,21 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
 // TODO see above
-const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
-    }
-
+const getAnonName = (firstName) => {
+  return new Promise((resolve,reject) =>{
     const fullName = `${firstName} Doe`;
-
-    callback(fullName);
-  }, 1000);
+    setTimeout(() => {
+      if (firstName) {
+        resolve(fullName)
```

Comment:
code style: because `fullName` is only used on this line, you could consider dropping it and constructing the full name directly. 

```suggestion
        resolve(`${firstName} Doe`)
```

Reply:
Also be aware that if you call resolve once this callback is completed it will continue execution, so this means that code that falls is defined underneath would be executed.

Now this isn't an easy as there's is nothing after this if else clause, but be aware

Reply:
Deleted the constant